--- a/.config/ags/modules/cheatsheet/keybinds.js
+++ b/.config/ags/modules/cheatsheet/keybinds.js
@@ -70,9 +70,9 @@ export default () => {
                 selectable: true,
                 justify: Gtk.Justification.CENTER,
                 className: 'txt-small txt',
-                label: 'Sheet data stored in <tt>~/.config/ags/modules/cheatsheet/data_keybinds.js</tt>\nChange keybinds in <tt>~/.config/hypr/hyprland/keybinds.conf</tt>'
+		label: 'Sheet data stored in <tt>~/.config/ags/modules/cheatsheet/data_keybinds.js</tt>\nChange Hyprland behavior in <tt>~/.config/hypr/custom/custom.conf</tt>'
             }),
             realKeybinds,
         ]
     })
-}
\ No newline at end of file
+}
