--- a/.config/hypr/custom/execs.conf
+++ b/.config/hypr/custom/execs.conf
@@ -1,2 +1,47 @@
-# You can make apps auto-start here
-# Relevant Hyprland wiki section: https://wiki.hyprland.org/Configuring/Keywords/#executing
+# Wallpaper
+#exec-once = swaybg -i ~/.config/eww/images/wallpaper/wallpaper
+#exec-once = swww init; swww kill; swww init # idk why tbh
+exec-once = swww-daemon --format xrgb
+ 
+# Status bar
+#exec-once = eww daemon && eww open bar && eww open bgdecor
+exec = ags &
+#exec-once = waybar
+
+# Input method
+exec-once = fcitx5
+exec-once = ~/.local/bin/initialSetup.sh
+exec = wlsunset -l 47.6 -L -122.3
+# Core components (authentication, lock screen, notification daemon)
+exec-once = dbus-update-activation-environment --all &
+exec-once = /usr/bin/gnome-keyring-daemon --start --components=secrets &
+exec-once = /usr/lib/polkit-gnome/polkit-gnome-authentication-agent-1 &
+#exec-once = swayidle -w timeout 3000 'swaylock' before-sleep 'swaylock' &
+#exec-once = swayidle -w timeout 4500 'pidof java || systemctl suspend' & # dont sleep if playing minecraft, else nvidia will fuck up
+exec-once = sleep 1 && dbus-update-activation-environment --systemd WAYLAND_DISPLAY XDG_CURRENT_DESKTOP # Some fix idk
+
+# Clipboard history
+exec-once= wl-paste --type text --watch cliphist store # Stores only text data
+exec-once= wl-paste --type image --watch cliphist store # Stores only image data
+exec-once = rm "$HOME/.cache/cliphist/db" # Clears history
+# Cursor
+exec-once = hyprctl setcursor Bibata-Modern-Classic 24
+
+# Plugins
+# are a good way to crash Hyprland
+
+# Gestures
+exec-once=touchegg
+
+# Bluetooth
+exec=blueman-tray &
+exec-once=sleep 4 && antimicrox --tray --hidden &
+
+# Allow GPU enabled XWayland applications that MUST run as root
+exec=xhost +SI:localuser:root
+
+# Network Manager
+exec-once=nm-applet &
+
+# Remote Desktop
+exec-once=~/.local/bin/sunshine &
