--- a/.config/hypr/custom/keybinds.conf
+++ b/.config/hypr/custom/keybinds.conf
@@ -1,2 +1,228 @@
-# You can put your preferred keybinds here
-# https://wiki.hyprland.org/Configuring/Binds/
\ No newline at end of file
+# $Secondary is a reference to Command or Win, depending on what is plugged into the computer.
+
+$Primary="Super"
+$Secondary="Control"
+$Tertiary="Shift"
+$Alternate="Alt"
+$MenuButton="Menu"
+
+# ################### It just works™ keybinds ###################
+# Volume
+bindl = ,XF86AudioMute, exec, pactl set-sink-mute @DEFAULT_SINK@ toggle
+bindle=, XF86AudioRaiseVolume, exec, wpctl set-volume @DEFAULT_AUDIO_SINK@ 5%+
+bindle=, XF86AudioLowerVolume, exec, wpctl set-volume @DEFAULT_AUDIO_SINK@ 5%-
+
+# Brightness
+# bindle=, XF86MonBrightnessUp, exec, light -A 5
+# bindle=, XF86MonBrightnessDown, exec, light -U 5
+bindle=, XF86MonBrightnessUp, exec, ags run-js 'brightness.screen_value += 0.05;'
+bindle=, XF86MonBrightnessDown, exec, ags run-js 'brightness.screen_value -= 0.05;'
+
+# ####################################### Applications ########################################
+# Apps: just normal apps
+# Code
+bind = $Primary$Secondary, U, exec, code
+# Discord
+bind = $Primary$Secondary, I, exec, armcord -enable-features=UseOzonePlatform --ozone-platform=wayland
+# Tidal
+bind = $Primary$Secondary, M, exec, kodi
+# foot
+bind = $Primary$Secondary, H, exec, foot
+bind = $Primary$Secondary$Tertiary, T, exec, foot sleep 0.01 && nmtui
+# finders
+bind = $Primary$Secondary, J, exec, thunar
+# firefox
+bind = $Primary$Secondary, L, exec, firefox
+bind = $Primary$Secondary$Tertiary, L, exec, /usr/bin/microsoft-edge-beta --enable-features=UseOzonePlatform --ozone-platform=wayland --gtk-version=4
+# notepad
+bind = $Primary$Secondary, 48, exec, subl
+bind = $Primary$Secondary, 47, exec, codium
+bind = $Primary$Secondary$Tertiary, 47, exec, jetbrains-fleet
+# office
+bind = $Secondary$Tertiary, W, exec, wps
+
+# calculator
+bind = $Primary$Secondary, 3, exec, ~/.local/bin/wofi-calc
+bind = ,XF86Calculator, exec ~/.local/bin/wofi-calc
+# Flux
+bind = $Primary$Secondary, N, exec, gammastep -O +3000 &
+bind = $Primary$Secondary$Alternate, N, exec, gammastep -0 +6500 &
+# Apps: Settings and config
+bind = $Secondary$Tertiary, 59, exec, XDG_CURRENT_DESKTOP="gnome" gnome-control-center
+bind = $Primary$Secondary, V, exec, pavucontrol 
+bind = $Primary$Tertiary, Home, exec, gnome-system-monitor
+bind = $Primary$Alternate, Insert, exec, foot -F btop
+
+# Actions
+bind = $Alternate, F4, killactive,
+bind = $Secondary$Alternate, Space, togglefloating, 
+bind = $Secondary$Alternate, Q, exec, hyprctl kill
+bind = $Primary$Tertiary$Alternate, Delete, exec, pkill wlogout || wlogout -p layer-shell
+bind = $Primary$Tertiary$Alternate$Secondary, Delete, exec, systemctl poweroff
+# bind = $Tertiary$Alternate,mouse_up, exec, wtype -M ctrl -k Prior
+# bind = $Tertiary$Alternate,mouse_down, exec, wtype -M ctrl -k Next
+
+# Screenshot, Record, OCR, Color picker, Clipboard history
+bind = $Secondary$Tertiary, D, exec,~/.local/bin/rubyshot | wl-copy
+bindl =,Print,exec,grim - | wl-copy
+bind = $Secondary$Tertiary, 4, exec, grim -g "$(slurp -d -c D1E5F4BB -b 1B232866 -s 00000000)" - | wl-copy
+bind = $Secondary$Tertiary, 5, exec, ~/.config/ags/scripts/record-script.sh --sound
+bind = $Secondary$Tertiary$Alternate, 5, exec, ~/.config/ags/scripts/record-script.sh --fullscreen-sound
+bind = $Primary$Tertiary, C, exec, wtype -M logo c -m logo && wl-paste >> ~/.config/snippets/active && notify-send "Snipped!"
+
+bind = $Secondary$Alternate, C, exec, hyprpicker -a
+#bind = $Secondary, V, exec, wtype -M ctrl v -m ctrl
+bind = $Primary$Alternate, Space, exec, cliphist list | wofi -Iim --dmenu | cliphist decode | wl-copy && wtype -M ctrl v -m ctrl
+bind = $Secondary$Alternate, V, exec, cliphist list | wofi -Iim --dmenu | cliphist decode | wl-copy && wtype -M ctrl v -m ctrl
+bind = $Primary, Menu, exec, tac ~/.local/share/snippets | wofi -Iim --dmenu | tr -d '\n' | wl-copy && wtype -M logo v -m logo
+bind = $Alternate, Menu, exec, wtype -M ctrl c -m ctrl && wl-paste >> ~/.local/share/snippets && notify-send "Added to snippets!"
+
+# Text-to-image
+# Normal
+bind = $Primary$Secondary$Tertiary,S,exec,grim -g "$(slurp -d -c D1E5F4BB -b 1B232866 -s 00000000)" "tmp.png" && tesseract "tmp.png" - | wl-copy && rm "tmp.png"
+# English
+bind = $Secondary$Tertiary,T,exec,grim -g "$(slurp -d -c D1E5F4BB -b 1B232866 -s 00000000)" "tmp.png" && tesseract -l eng "tmp.png" - | wl-copy && rm "tmp.png"
+# Japanese
+bind = $Secondary$Tertiary,J,exec,grim -g "$(slurp -d -c D1E5F4BB -b 1B232866 -s 00000000)" "tmp.png" && tesseract -l jpn "tmp.png" - | wl-copy && rm "tmp.png"
+
+# Media
+bind = $Secondary$Tertiary, N, exec, playerctl next || playerctl position `bc <<< "100 * $(playerctl metadata mpris:length) / 1000000 / 100"`
+bindl  = , XF86AudioNext,           exec, playerctl next 
+bindl  = , XF86AudioPrev,           exec, playerctl previous
+bindl  = , XF86AudioPlay,           exec, playerctl play-pause
+bind = $Secondary$Tertiary, B, exec, playerctl previous
+bind = $Secondary$Tertiary, P, exec, playerctl play-pause
+
+#Lock screen  |  blur: --effect-blur=20x202
+bind = $Secondary, L, exec, swaylock
+bind = $Secondary$Tertiary, L, exec, swaylock
+# bindl = $Secondary$Tertiary, L, exec, sleep 0.1 && systemctl suspend
+
+# App launcher
+bind = $Primary$Secondary, Slash, exec, pkill anyrun || anyrun
+
+# ##################################### AGS keybinds #####################################
+bindr = $Primary$Secondary, R, exec, hyprctl reload; pkill ags activewin.sh activews.sh gohypr bash ydotool; ags &
+bind = $Primary$Secondary, T, exec, ~/.config/ags/scripts/color_generation/switchwall.sh
+bind = $Alternate, Tab, exec, ags -t 'overview'
+bind = $Secondary, Space, exec, ags -t 'overview'
+bind = $Secondary$Alternate, Slash, exec, ags -t 'cheatsheet0'
+bind = $Secondary, B, exec, ags -t 'sideleft'
+bind = $Secondary, N, exec, ags -t 'sideright'
+bind = $Secondary, M, exec, ags run-js 'openMusic$Primarys.value = (!Mpris.getPlayer() ? false : !openMusic$Primarys.value);'
+bind = $Secondary, Comma, exec, ags run-js 'openColorScheme.value = true; Utils.timeout(2000, () => openColorScheme.value = false);'
+bind = $Secondary, K, exec, ags -t 'osk'
+bind = $Primary$Alternate, Delete, exec, ags -t 'session'
+bindr = $Primary, $Primary_R, exec, ags run-js 'indicator.popup(-1);'
+bindr = $Primary, $Primary_R, exec, ags run-js 'Notifications.notifications.forEach((notif) => notif.dismiss())'
+bindle = , XF86AudioRaiseVolume, exec, ags run-js 'indicator.popup(1);'
+bindle = , XF86AudioLowerVolume, exec, ags run-js 'indicator.popup(1);'
+bindle = , XF86MonBrightnessUp, exec, ags run-js 'indicator.popup(1);'
+bindle = , XF86MonBrightnessDown, exec, ags run-js 'indicator.popup(1);'
+bindr = $Primary, $Primary_R, exec, ags run-js 'App.closeWindow("sideright");'
+bindr = $Primary, $Primary_R, exec, ags run-js 'App.closeWindow("cheatsheet");'
+bindr = $Primary, $Primary_R, exec, ags run-js 'App.closeWindow("osk");'
+bindr = $Primary, $Primary_R, exec, ags run-js 'App.closeWindow("session");'
+bindr = $Primary, $Primary_R, exec, ags run-js 'openMusic$Primarys.value = false'
+bindr = $Primary, $Primary_R, exec, ags run-js 'openColorScheme.value = false'
+
+# ##################################### Plugins #########################################
+bind = $Primary$Secondary, P, exec, hyprctl plugin load "~/.config/hypr/plugins/droidbars.so"
+bind = $Primary$Secondary, O, exec, hyprctl plugin unload "~/.config/hypr/plugins/droidbars.so"
+
+# Testing
+# bind = $Secondary$Alternate, f12, exec, notify-send "Hyprland version: $(hyprctl version | head -2 | tail -1 | cut -f2 -d ' ')" "owo" -a 'Hyprland keybind'
+bind = $Secondary$Alternate, f12, exec, notify-send "Millis since epoch" "$(date +%s%N | cut -b1-13)" -a 'Hyprland keybind'
+bind = $Secondary$Alternate, Equal, exec, notify-send "Urgent notification" "Ah hell no" -u critical -a 'Hyprland keybind'
+
+# ########################### Keybinds for Hyprland ############################
+# Swap windows
+bind = $Secondary$Tertiary, left, movewindow, l
+bind = $Secondary$Tertiary, right, movewindow, r
+bind = $Secondary$Tertiary, up, movewindow, u
+bind = $Secondary$Tertiary, down, movewindow, d
+# Move focus
+bind = $Secondary, left, movefocus, l
+bind = $Secondary, right, movefocus, r
+bind = $Alternate, up, movefocus, u
+bind = $Alternate, down, movefocus, d
+bind = $Secondary, BracketLeft, movefocus, l
+bind = $Secondary, BracketRight, movefocus, r
+
+# Workspace, window, tab switch with keyboard
+bind = $Primary$Secondary, right, workspace, +1
+bind = $Primary$Secondary, left, workspace, -1
+bind = $Primary$Secondary, BracketLeft, workspace, -1
+bind = $Primary$Secondary, BracketRight, workspace, +1
+bind = $Primary$Secondary, up, workspace, -5
+bind = $Primary$Secondary, down, workspace, +5
+bind = $Secondary, Page_Down, workspace, +1
+bind = $Secondary, Page_Up, workspace, -1
+bind = $Primary$Secondary, Page_Down, workspace, +1
+bind = $Primary$Secondary, Page_Up, workspace, -1
+bind = $Secondary$Alternate, Page_Down, movetoworkspace, +1
+bind = $Secondary$Alternate, Page_Up, movetoworkspace, -1
+bind = $Secondary$Tertiary, Page_Down, movetoworkspace, +1
+bind = $Secondary$Tertiary, Page_Up, movetoworkspace, -1
+bind = $Primary$Secondary$Tertiary, Right, movetoworkspace, +1
+bind = $Primary$Secondary$Tertiary, Left, movetoworkspace, -1
+bind = $Secondary$Tertiary, mouse_down, movetoworkspace, -1
+bind = $Secondary$Tertiary, mouse_up, movetoworkspace, +1
+bind = $Secondary$Alternate, mouse_down, movetoworkspace, -1
+bind = $Secondary$Alternate, mouse_up, movetoworkspace, +1
+
+# Window split ratio
+binde = $Primary$Secondary, Minus, splitratio, -0.1
+binde = $Primary$Secondary, Equal, splitratio, 0.1
+binde = $Secondary, Semicolon, splitratio, -0.1
+binde = $Secondary, Apostrophe, splitratio, 0.1
+
+# Fullscreen
+bind = $Primary$Secondary, F, fullscreen, 0
+bind = $Primary$Secondary, D, fullscreen, 1
+bind = $Secondary$Alternate, F, fakefullscreen, 0
+
+# Switching
+bind = $Secondary, 1, workspace, 1
+bind = $Secondary, 2, workspace, 2
+bind = $Secondary, 3, workspace, 3
+bind = $Secondary, 4, workspace, 4
+bind = $Secondary, 5, workspace, 5
+bind = $Secondary, 6, workspace, 6
+bind = $Secondary, 7, workspace, 7
+bind = $Secondary, 8, workspace, 8
+bind = $Secondary, 9, workspace, 9
+bind = $Secondary, 0, workspace, 10
+bind = $Primary$Secondary, S, togglespecialworkspace,
+bind = $Alternate, Tab, cyclenext
+bind = $Alternate, Tab, bringactivetotop,   # bring it to the top
+
+# Move window to workspace $Secondary + $Alternate + [0-9] 
+bind = $Secondary $Alternate, 1, movetoworkspacesilent, 1
+bind = $Secondary $Alternate, 2, movetoworkspacesilent, 2
+bind = $Secondary $Alternate, 3, movetoworkspacesilent, 3
+bind = $Secondary $Alternate, 4, movetoworkspacesilent, 4
+bind = $Secondary $Alternate, 5, movetoworkspacesilent, 5
+bind = $Secondary $Alternate, 6, movetoworkspacesilent, 6
+bind = $Secondary $Alternate, 7, movetoworkspacesilent, 7
+bind = $Secondary $Alternate, 8, movetoworkspacesilent, 8
+bind = $Secondary $Alternate, 9, movetoworkspacesilent, 9
+bind = $Secondary $Alternate, 0, movetoworkspacesilent, 10
+bind = $Primary$Tertiary$Secondary, Up, movetoworkspacesilent, special
+bind = $Secondary$Alternate, S, movetoworkspacesilent, special
+
+# Scroll through existing workspaces with ($Primary) + $Secondary + scroll
+bind = $Secondary, mouse_up, workspace, +1
+bind = $Secondary, mouse_down, workspace, -1
+bind = $Primary$Secondary, mouse_up, workspace, +1
+bind = $Primary$Secondary, mouse_down, workspace, -1
+
+# Move/resize windows with $Secondary + LMB/RMB and dragging
+bindm = $Secondary, mouse:273, resizewindow
+bindm = $Primary$Secondary, mouse:273, resizewindow
+bindm = ,mouse:274, movewindow
+bindm = $Primary$Secondary, Z, movewindow
+bind = $Primary$Secondary, Backslash, resizeactive, exact 640 480
+
+bind = $Secondary$Alternate, J, exec, ydotool key 105:1 105:0 
+
