--- a/src/keys.c
+++ b/src/keys.c
@@ -15,7 +15,7 @@ const struct modifier modifiers[MAX_MOD] = {
 	{MOD_CTRL, KEYD_LEFTCTRL},
 };
 
-const struct keycode_table_ent keycode_table[256] = {
+const struct keycode_table_ent keycode_table[465] = {
 	[KEYD_ESC] = { "esc", "escape", NULL },
 	[KEYD_1] = { "1", NULL, "!" },
 	[KEYD_2] = { "2", NULL, "@" },
@@ -376,7 +376,7 @@ int parse_key_sequence(const char *s, uint8_t *codep, uint8_t *modsp)
 		c += 2;
 	}
 
-	for (i = 0; i < 256; i++) {
+	for (i = 0; i < 465; i++) {
 		const struct keycode_table_ent *ent = &keycode_table[i];
 
 		if (ent->name) {
